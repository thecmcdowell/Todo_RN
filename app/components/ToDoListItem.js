import React from "react";
import { View, Text, StyleSheet } from "react-native";

const ToDoItem = item => {
  let task = item.item.item;
  if (task.completed) {
    return (
      <View style={[styles.item, { backgroundColor: "#a2d246" }]}>
        <Text style={styles.text}>{task.name}</Text>
      </View>
    );
  } else {
    return (
      <View style={[styles.item, { backgroundColor: "#007acc" }]}>
        <Text style={styles.text}>{task.name}</Text>
      </View>
    );
  }
};

const styles = StyleSheet.create({
  item: {
    justifyContent: "center",
    height: 30,
    borderColor: "gray",
    borderWidth: 0.5,
    borderRadius:10,
    minWidth: "90%",
    marginBottom: 10
  },
  text: {
    alignSelf: 'center',
    color: '#F5FCFF'
  }
});

export default ToDoItem;
