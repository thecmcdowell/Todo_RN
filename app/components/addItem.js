import React from "react";
import { TextInput, Button, View, StyleSheet } from "react-native";
import { connect } from "react-redux";
import { addTodo } from "../actions/todoActions";

class AddTask extends React.Component {
  state = {
    text: ""
  };

  addTask = () => {
    if (this.state.text !== "") {
      this.props.addTodo(this.state.text);
      this.setState({ data: this.state.todos });
      this.setState({ text: "" });
    }
  };

  render() {
    return (
      <View style={{ paddingTop: 10, paddingBottom: 50 }}>
        <TextInput
          placeholder="Add ToDo"
          value={this.state.text}
          onChangeText={text => this.setState({ text })}
          returnKeyType="done"
          onSubmitEditing={this.addTask.bind(this, this.state.text)}
          style={styles.input}
        />
        <Button
          onPress={this.addTask.bind(this, this.state.text)}
          title="Add Task"
          color= "#a1fd9c"
          disabled={this.state.text ? false: true}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  input: {
    height: 40,
    borderColor: "gray",
    backgroundColor: "white",
    borderWidth: 1,
    paddingLeft: 10,
    paddingRight: 10,
    width: 150
  }
});

export default connect(
  null,
  { addTodo }
)(AddTask);
