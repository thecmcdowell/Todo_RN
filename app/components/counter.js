import React from "react";
import { Text } from "react-native";

const Counter = todos => {
  let completed = todos.todos.filter(todo => todo.completed === false);
  if (completed == 0) {
    return (
      <Text style={{ fontWeight: "bold", fontSize: 25, color:"#F5FCFF"}}>
        You have nothing to do! You did all of it! ⭐️
      </Text>
    );
  } else {
    return (
      <Text style={{ fontSize: 25,  color:"#F5FCFF"}}>
        You Have {completed.length} task(s) remaining!
      </Text>
    );
  }
};

export default Counter