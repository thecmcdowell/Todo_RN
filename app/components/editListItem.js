import React from "react";
import {
  Modal,
  View,
  TextInput,
  Button,
  Text,
  SafeAreaView
} from "react-native";
import { connect } from "react-redux";
import { deleteTodo, editTodo, toggleComplete } from "../actions/todoActions";

class EditItem extends React.Component {
  state = {
    text: "",
    visible: false
  };
  editTask = () => {
    let item = this.props.item.item;
    let updatedItem = {
      name: this.state.text,
      id: item.id,
      completed: item.completed
    };
    this.props.editTodo(item, updatedItem);
    this.setState({ text: "" });
    this.props.dismiss();
  };
  toggleComplete = () => {
    let item = this.state.selectedItem;
    this.props.toggleComplete(this.props.item);
    this.props.dismiss();
  };

  deleteTask = () => {
    let item = this.props.item.item;
    this.props.deleteTodo(item.id);
    this.props.dismiss();
  };
  render() {
    let item = this.props.item.item;
    if (!item) {
      return null;
    } else {
      let completed = this.props.item.item.completed;
      return (
        <SafeAreaView>
          <Modal
            transparent={false}
            animationType="slide"
            visible={this.props.visible}
            style={{ minHeight: "100%", backgroundColor: "#4F6D7A" }}
          >
            <View style={{ minHeight: "100%", backgroundColor: "#4F6D7A" }}>
              <View style={{ paddingTop: 100, alignSelf: "center" }}>
                <Text
                  style={{
                    alignSelf: "center",
                    color: "#F5FCFF",
                    fontSize: 25
                  }}
                >
                  {item.name}
                </Text>
                <Text
                  style={[
                    styles.status,
                    {fontSize: 15}
                  ]}
                >
                  {completed ? (
                    <Text>Completed!</Text>
                  ) : (
                    <Text>Still Needs doing.</Text>
                  )}
                </Text>
                <View style={{ flexDirection: "row" }}>
                  <TextInput
                    placeholder={item.name}
                    value={this.state.text}
                    onChangeText={text => this.setState({ text })}
                    returnKeyType="done"
                    style={styles.input}
                  />
                  <Button
                    disabled={this.state.text === ""}
                    title="Edit Task"
                    onPress={() => this.editTask()}
                    color="#a1fd9c"
                  />
                </View>
                <Button
                  color="#a1fd9c"
                  title={completed ? `Didn't finish?` : "Completed?"}
                  onPress={() => this.toggleComplete()}
                />
              </View>
              <View>
                <Button
                  color="#a1fd9c"
                  title="Delete"
                  onPress={() => this.deleteTask()}
                />
                <Button
                  color="#a1fd9c"
                  title="Cancel"
                  onPress={() => this.props.dismiss()}
                />
              </View>
            </View>
          </Modal>
        </SafeAreaView>
      );
    }
  }
}

const styles = {
  input: {
    height: 40,
    borderColor: "gray",
    borderWidth: 1,
    paddingLeft: 10,
    paddingRight: 10,
    width: 150,
    backgroundColor: "white",
  },
  main: {
    backgroundColor: "#88d6f2"
  },
  status: {
    alignSelf: "center",
    color: "#F5FCFF",
    fontSize: 15
  }
};

export default connect(
  null,
  { deleteTodo, editTodo, toggleComplete }
)(EditItem);
