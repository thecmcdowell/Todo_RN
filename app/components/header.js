import React from "react";
import { Text, View } from "react-native";

const Header = () => {
  return (
    <View style={style.header}>
    <View>
      <Text style={style.text}> Greens are finished! Blues aren't finshed ðŸ˜Ÿ</Text>
      <Text style={style.text}> Tap on an item to see your options</Text>
      </View>
    </View>
  );
};

const style = {
  header: {
    marginBottom: 10,
    borderColor: "gray",
    borderWidth: 1
  },
  text: {
    color: "#F5FCFF"
  }
};

export default Header