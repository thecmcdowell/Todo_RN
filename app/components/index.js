import ToDoItem from './ToDoListItem'
import EditItem from './editListItem'
import Counter from './counter'
import AddTask from './addItem'
import Header from './header'

export {
    ToDoItem,
    EditItem,
    Counter,
    AddTask,
    Header
}