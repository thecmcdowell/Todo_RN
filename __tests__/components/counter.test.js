import 'react-native'
import React from 'react'
import {Counter} from '../../app/components/index'

import renderer from 'react-test-renderer'
describe('it renders counter', () => {
    it('renders counter with no unfinished tasks', () => {
        const todos = []
        const tree = renderer.create(
            <Counter todos={todos} />
        )
        expect(tree).toMatchSnapshot()
    }),
    it('renders counter with unfinished tasks', () => {
        const todos = [{
            name: 'write tests',
            index: 'todos_1',
            completed: false
        }]
        const tree = renderer.create(
            <Counter todos={todos} />
        )
        expect(tree).toMatchSnapshot()
    })
})
