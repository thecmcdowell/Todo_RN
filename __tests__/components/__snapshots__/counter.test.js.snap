// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders counter renders counter with no unfinished tasks 1`] = `
<Text
  style={
    Object {
      "color": "#F5FCFF",
      "fontSize": 25,
      "fontWeight": "bold",
    }
  }
>
  You have nothing to do! You did all of it! ⭐️
</Text>
`;

exports[`it renders counter renders counter with unfinished tasks 1`] = `
<Text
  style={
    Object {
      "color": "#F5FCFF",
      "fontSize": 25,
    }
  }
>
  You Have 
  1
   task(s) remaining!
</Text>
`;
